### PR TITLE
Added EnergyAttibuteProvider

### PR DESCRIPTION
**Pull request by falseresync**

### CODE REVIEW

File: src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java
Comment (LemmaEOF) [inline src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java]: This, sadly, won't work. An Item is, for all intents and purposes, static; you never store any info on the item itself. I think it'd be best if this was passed a compound tag, so that it could be used on both items and entities simultaneously. Also, once it's merged, I'll write some javadocs for it.

**Review by LemmaEOF — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java
Comment (falseresync) [reply src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java]: Like add a to/fromTag methods and a simple impl for them?

Comment (LemmaEOF) [reply src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java]: Yeah, instead of `EnergyAttribute getEnergy()` do `EnergyAttribute fromTag(CompoundTag tag)` and `void toTag(EnergyAttribute attribute, CompoundTag tag)`.

Comment (falseresync) [reply src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java]: I think I'll leave the getEnergy() as an easy-to-use one for entities

Comment (falseresync) [reply src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java]: Should it have default impls?

Comment (LemmaEOF) [reply src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java]: I don't think so.

Comment (falseresync) [reply src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java]: Don't merge yet, one more change

**Review by LemmaEOF — APPROVED** [review]
I'm alright with this, but I just wanna make sure it looks good to @falkreon before I merge, since he's got a better eye for API design than I do.

File: src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java
Comment (Juuxel) [inline src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java]: This could conflict with a vanilla `void fromTag(CompoundTag)` if implemented on some type that has that. (Edit: Not sure if entities do)

Comment (falseresync) [reply src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java]: It shouldn’t. It would just require it.

Comment (LemmaEOF) [reply src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java]: I think it might cause an issue, yeah. Method signatures don't include the return type.

Comment (falseresync) [reply src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java]: Ohh, right. But I’m away from my PC by now again. I will solve it tomorrow by my time, or you may if you want to

**Comment by falkreon** [comment]
I'm in favor of removing the Silk dependency.

That said, LBA is the correct way to discover an EnergyAttribute, and even if it wasn't, this PR is dangerous because of the fromTag and toTag methods. Adding a BlockEntity interface API will just make discovery *harder*.

**Review by falkreon — CHANGES_REQUESTED** [review]
I would prefer to either retarget this PR to removing silk and killing the transient dependencies for LBA, but get rid of the interface the PR was made for; or,

to close the PR altogether but accomplish the aforementioned through direct changes or their own PR.

**Review by falkreon — APPROVED** [review]
This settles all my concerns. It's ready for merge.

**Pull request merged**
